src/cli/bootupctl.rs

```diff
@@ -101,38 +101,54 @@ impl CtlCommand {
             bootupd::print_status(&r)?;
         }
 
-        client.shutdown()?;
         Ok(())
     }
 
     /// Runner for `update` verb.
     fn run_update() -> Result<()> {
-        let mut client = ClientToDaemonConnection::new();
-        client.connect()?;
-
-        bootupd::client_run_update(&mut client)?;
-
-        client.shutdown()?;
-        Ok(())
+        let running_in_systemd = running_in_systemd();
+        if !running_in_systemd {
+            let r = Command::new("systemd-run")
+                .args(BOOTUPD_ARGS)
+                .arg("update")
+                .exec();
+            // If we got here, it's always an error
+            return Err(r.into());
+        }
```

Comment:
Let's factor this out as a helper `fn ensure_running_in_systemd() -> Result<()>` or so.